### DISCUSSION TITLE
Internationalization for egui

### DISCUSSION THREAD

**Discussion by John-Nagle** [emilk/egui#1751] [Q&A]
Anyone have a good internationalization solution for egui? Something where UI functions that take a string argument get something like 

t!("menu.open",lang)

and "menu.open" is looked up in a file of translations?

The problem is that it has to be really fast after the first lookup, because this runs on every frame. I've tried a few crates, but either the lookup is at run time (fluent) or the tricks to do it at compile time don't really work right (internationalization). Suggestions?

**Comment by hydra** [comment]
I'm a bit late to the party, but was this crate one you tried https://github.com/fewensa/egui-i18n ?  if so did you encounter any issues?

**Comment by John-Nagle** [comment]
I wrote my own, years ago: https://github.com/John-Nagle/ui-mock/blob/main/src/basicintl.rs

**Comment by YgorSouza** [comment]
FWIW I use [i18n-embed-fl](https://crates.io/crates/i18n-embed-fl) in some localized apps I work on with egui, and it works great. See the "convenience macro" at the bottom of the readme. It's just `fl!("message-id", arg = "value")` in place of the string, then put all your localization files in a folder that is embedded statically into the app, and create a static for the loader that you can call to change the language whenever you want, and since it's immediate UI, everything updates nicely unless you have strings manually cached somewhere. It's all checked at compile time too.
